pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -195,6 +195,9 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 		updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
 		return reconcile.Result{}, handleError("Failed to fetch the Build instance", err, updateErr)
 	}
+	if build.Status.Registered != corev1.ConditionTrue {
+		return reconcile.Result{}, fmt.Errorf("Build is not ready, registered status: %v, reason: %v", build.Status.Registered, build.Status.Reason)
```

Comment:
Should we also update the buildRun status to show something like: `The Build is not registered correctly, etc...` to notify the user there is a problem on Build side?

Reply:
Oh yeah, +1 from me. Should be something like:

```go
		updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, fmt.Printf("Build is not ready, registered status: %v, reason: %v", build.Status.Registered, build.Status.Reason))
		return reconcile.Result{}, handleError("Build is not ready", err, updateErr)

```